### PR TITLE
feat: add a11y/permitted-contents — content-model membership over the element tree

### PR DESCRIPTION
**Pull request by oekazuma**
## Summary

Implements the "Build" decision of the permitted-contents measurement (#545), per the Fable-approved design in `docs/superpowers/specs/2026-08-20-permitted-contents-rule-design.md`.

- **New rule `a11y/permitted-contents`** (95th rule): every literal child element must be a member of its literal parent's permitted-content set, per the vendored HTML content models. Membership only — order and count are statically unjudgeable; child attribution looks through control-flow blocks; transparent elements (`<a>`, `<ins>`, …) pass the question up to the nearest opaque ancestor; the `dl > div` wrapper and the other conditional models are honoured.
- **Severity split by consequence** (per-result, a first — `ComponentIssue` gains an optional `severity`): broken structure (closed containers like `<ul>`/`<table>`, headings crossing the boundary, structure-bound tags like `<li>` outside a list) is `warning`; category mismatches (`<button><div>`) are `info` — the measured 234-finding benign class must not gate.
- **Deliberate non-verdicts**, each measured and recorded: `<option>` rich content (compiler wins), interactive nesting (`a11y/interactive-nesting` owns it — one defect is never two findings), canvas fallback (WHATWG permits interactive fallback; the dataset is stricter), everything across a component/snippet/slot/`{@render}`/`{@html}`/custom-element boundary, SVG subtrees, text children.
- **Fact plumbing**: `ElementFact` (internal) gains `parent` links (chain broken at every non-lexical construct), literal attribute `value`s, `hasSpread`, and `unknownContent`; the evaluator (`html-spec/content-model.ts`) ports the measurement probe's reviewed semantics. No new I/O — rides the existing parse pass.
- **Corpus cross-check**: AdventureLog matches the probe exactly (91); networking-toolbox reports 193 vs the probe's 152, reconciled exactly — the CLI's SCSS-tolerant parser analyzes the 31 files the probe's raw parser rejected (recorded in the measured doc).
- Guards: 17 evaluator/rule unit tests (the probe's self-test cases), kitchen-sink plants for both severity classes with an e2e assertion that the severities differ, docs en/ja (+ mode differences) stamped, rules-index/skills/CLI-snapshot regens, changeset (minor ×3).

Two Fable review rounds (design: 3 rounds to APPROVE; implementation: APPROVE with SHOULDs, all applied — canvas carve-out, custom-element chain breaks, `video[src]` fixtures, camelCase SVG key lookup).

## Verify

`pnpm build && pnpm typecheck && pnpm lint && pnpm -r test && pnpm smoke && pnpm check:publish && pnpm --filter docs run translate:check` — all green.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

- **New Features**
  - Added the `a11y/permitted-contents` rule to detect invalid parent-child HTML relationships.
  - Reports warnings or informational notices based on violation impact.
  - Supports conditional, transparent, custom, SVG, and unknown content handling.

- **Documentation**
  - Added English and Japanese guidance, catalog entries, remediation advice, and examples.

- **Tests**
  - Added coverage for valid content, severity levels, edge cases, and end-to-end reporting.

### CODE REVIEW

File: docs/src/content/docs/rules/a11y/permitted-contents.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+---
+
+**Severity:** warning (broken structure) / info (category mismatches) — see below
+
+## What it checks
+
+Every literal child element must be a member of its literal parent's permitted-content set, per
+the HTML content models (the same vendored spec data as the other element rules). Membership
+only — never order or count, which components and `{#if}`/`{#each}` blocks make unjudgeable — and
+a child inside a control-flow block still counts as the enclosing element's child. Transparent
```
[the remark this answers is not in the document]
Comment (oekazuma) [reply docs/src/content/docs/rules/a11y/permitted-contents.md]: Fixed in 3ad26e2f — the sentence now reads "never order or count, because components and `{#if}`/`{#each}` blocks make those impossible to judge statically" (en + ja, stamped).

File: packages/core/src/component-parse.ts [hunk 2]
```diff
@@ -18,6 +18,11 @@ import type {
   UnnamedInteractiveFact
 } from './component.js';
 import { isRootRelativePath } from './base-path.js';
+import { HTML_SPEC } from './html-spec/generated.js';
+
+// Tag names the spec data knows, lowercased — 32 SVG keys are camelCase (`svg:clipPath`), and the
+// collector lowercases tags, so a direct key lookup would misread those as unknown tags.
+const KNOWN_TAGS = new Set(Object.keys(HTML_SPEC.elements).map((k) => k.replace(/^svg:/, '').toLowerCase()));
```
[the remark this answers is not in the document]
Comment (oekazuma) [reply packages/core/src/component-parse.ts:25]: Declining this one, with reasoning. The unified lookup deliberately matches the measurement probe (`SPECS.has(tag) || SPECS.has('svg:'+tag)` — docs/superpowers/specs/2026-08-20-permitted-contents-measured.md), so the shipped numbers stay reconciled with the measured ones. On the concrete case: a literal lowercase `<circle>` outside `<svg>` cannot be a Svelte component (components are capitalized) and is genuinely invalid HTML — the browser treats it as an unknown element — so `div > circle` being reported is a true finding, not a false one, consistent with the repo's namespace-blind precedent (accessible-name/interactive-nesting). As a parent, an SVG-only tag has no `HTML_SPEC.elements` entry under its bare name, so `judgeContent` returns no-spec and stays silent — no HTML model is ever applied to an SVG-only parent. The corpus had zero occurrences of either shape.

**Comment by oekazuma** [comment]
@coderabbitai review

**Comment by oekazuma** [comment]
@coderabbitai review
